### ISSUE TITLE
Multiline comments do not work within @()

### ISSUE THREAD

**Issue by battaglia01** [xonsh/xonsh#2721]
Example:

```
$ print("""Line
break""")
Line
break

$ echo @("""Line
break""")
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/xonsh/ptk/shell.py", line 137, in _push
    locs=None)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 20054, in compile
    transform=transform)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 20023, in parse
    tree, input = self._parse_ctx_free(input, mode=mode, filename=filename)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 20123, in _parse_ctx_free
    raise original_error from None
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 20110, in _parse_ctx_free
    debug_level=(self.debug_level > 2))
  File "/usr/local/lib/python3.5/dist-packages/xonsh/parsers/base.py", line 349, in parse
    tree = self.parser.parse(input=s, lexer=self.lexer, debug=debug_level)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/ply/ply/yacc.py", line 331, in parse
    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/ply/ply/yacc.py", line 1199, in parseopt_notrack
    tok = call_errorfunc(self.errorfunc, errtoken, self)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/ply/ply/yacc.py", line 193, in call_errorfunc
    r = errorfunc(token)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/parsers/base.py", line 2760, in p_error
    column=p.lexpos))
  File "/usr/local/lib/python3.5/dist-packages/xonsh/parsers/base.py", line 482, in _parse_error
    raise err
  File "<string>", line None
SyntaxError: <xonsh-code>:1:5: ('code: @(',)
echo @("""Line
     ^
```

Adding a backslash pre-line break prevents a crash, but as a result doesn't parse the newline (which is the correct pythonic behavior)

```
$ print("""No line\
break""")
No linebreak

$ echo @("""No line\
break""")
No linebreak
```

My xonfig is:

```
+------------------+----------------------+
| xonsh            | 0.7.0                |
| Git SHA          | 3779e8b5             |
| Commit Date      | Jul 14 14:40:04 2018 |
| Python           | 3.5.2                |
| PLY              | 3.9                  |
| have readline    | True                 |
| prompt toolkit   | 1.0.15               |
| shell type       | prompt_toolkit1      |
| pygments         | None                 |
| on posix         | True                 |
| on linux         | True                 |
| distro           | neon                 |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

**Comment by scopatz** [comment]
Thanks for pointing this out @battaglia01! Note that it does work, if you provide an explicit subprocess:

```console
$ ![echo @("""hello 
  mom""")]
hello 
mom
```

But this is certainly  a bug.

**Comment by battaglia01** [comment]
Also, strangely enough, this works:

```
$ echo """lol
asd"""
lol
asd
```

**Comment by scopatz** [comment]
Yeah strings in subproc mode act the same way they do in Python mode

**Comment by scopatz** [comment]
Potential fix available in #2733!  Great job finding this @battaglia01!
